src/xcb_ffi/mod.rs

```diff
@@ -212,19 +212,27 @@ impl XCBConnection {
                 )
             }
         } else {
-            // Convert the FDs into an array of ints. libxcb will close the FDs.
-            let mut fds: Vec<_> = fds.into_iter().map(RawFdContainer::into_raw_fd).collect();
-            let num_fds = fds.len().try_into().unwrap();
-            let fds_ptr = fds.as_mut_ptr();
-            unsafe {
-                raw_ffi::xcb_send_request_with_fds64(
-                    self.conn.as_ptr(),
-                    flags,
-                    &mut new_bufs_ffi[2],
-                    &protocol_request,
-                    num_fds,
-                    fds_ptr,
-                )
+            #[cfg(unix)]
+            {
+                // Convert the FDs into an array of ints. libxcb will close the FDs.
+                let mut fds: Vec<_> = fds.into_iter().map(RawFdContainer::into_raw_fd).collect();
+                let num_fds = fds.len().try_into().unwrap();
+                let fds_ptr = fds.as_mut_ptr();
+                unsafe {
+                    raw_ffi::xcb_send_request_with_fds64(
+                        self.conn.as_ptr(),
+                        flags,
+                        &mut new_bufs_ffi[2],
+                        &protocol_request,
+                        num_fds,
+                        fds_ptr,
+                    )
+                }
+            }
+            #[cfg(not(unix))]
+            {
+                // It is not possible to create a `RawFdContainer` on non-unix.
+                unreachable!();
```

Comment:
Could you move the comment (or something like it) into the argument of `unreachable!`? (I think `unreachable!("foo")` should work, does it?)

Reply:
Done